### ISSUE TITLE
IDE0005 RemoveUnnecessaryImportsFixable removes Import needed to resolve ambiguity

### ISSUE THREAD

**Issue by jrmoreno1** [dotnet/roslyn#63591]
### Analyzer

IDE000 Imports statement is unnecessary 

Version:
VS 2022 17.3.2

### Steps To Reproduce
Have class that has a name space collision that is resolved by including an import that is otherwise unused.  Import is removed, BC30561 appears.

### Expected behavior
Namespace is neither flagged nor removed.

### Actual behavior
Namespace is flagged (lightbulb) and removed for document/project/solution.

## Additional context
Project is a Framework 4.8 non-sdk project.

**Comment by CyrusNajmabadi** [comment]
@jrmoreno1 can you provide a small repro?

**Comment by jrmoreno1** [comment]
@CyrusNajmabadi : Yes, I can, just managed to do so while figuring out what is probably the problem: My main project has an Imports for the Namespace in the .vbproj file.  If I remove that it no longer shows the Import as being unnecessary.

[RemoveUnnecessaryImportsFixable.zip](https://github.com/dotnet/roslyn/files/9427492/RemoveUnnecessaryImportsFixable.zip)
.

**Comment by CyrusNajmabadi** [comment]
@jrmoreno1 Thanks.  repros for me as well.  @jaredpar looks to be a compiler issue.  Compiler is reporting that the import is unnecessary. However, removal of it does lead to the ambiguity warning.

**Comment by CyrusNajmabadi** [comment]
Removing @akhera99 as we are only operating on the reported compiler diagnostics.

**Comment by AlekseyTs** [comment]
@CyrusNajmabadi 

I do not see compiler reporting ERRID.HDN_UnusedImportStatement or ERRID.HDN_UnusedImportClause for the MainProject\Example.aspx.vb file, but ide shows IDE0005 for it. I do see  ERRID.HDN_UnusedImportStatement reported for generated files. Like AssemblyInfo.vb and MainProject\obj\Debug\.NETFramework,Version=v4.8.AssemblyAttributes.vb. Is it possible that IDE reports the error for the wrong file? Or, perhaps, the file is included into more than one compilation?

**Comment by AlekseyTs** [comment]
Possibly ``ExternalSource``` directive is used somewhere?

**Comment by AlekseyTs** [comment]
Here is a simple test case that is supposed to demonstrate the problem, but no repro:
```
        <Fact, WorkItem(63591, "https://github.com/dotnet/roslyn/issues/63591")>
        Public Sub AmbiguityDueToProjectLevelImports()
            Dim source =
<compilation>
    <file name="a.vb"><![CDATA[
Imports Ns1

Public Class Program
    Inherits Ns2.C2

    Public Sub Main()
        Dim x = New C1()
    End Sub

End Class

Namespace Global.Ns1
    Public Class C1
    End Class
End Namespace

Namespace Global.Ns2
    Public Class C1
    End Class

    Public Class C2
    End Class
End Namespace
]]></file>
</compilation>
            Dim compilation = CreateCompilation(source, options:=TestOptions.DebugDll.WithGlobalImports(GlobalImport.Parse({"Ns1", "Ns2"})).WithRootNamespace("MainClass"))
            compilation.AssertTheseDiagnostics(
<errors>
</errors>,
                suppressInfos:=False)
        End Sub
```

**Comment by jrmoreno1** [comment]
That's in a different namespace, but it doesn't look like it is in a separate project.

**Comment by jrmoreno1** [comment]
Changed project to Console app in order to reduce the noise.  Still has the same problem.

[RemoveUnnecessaryImportsFixableConsole.zip](https://github.com/dotnet/roslyn/files/9434626/RemoveUnnecessaryImportsFixableConsole.zip)
.

**Comment by jrmoreno1** [comment]
@AlekseyTs : I have reduced the example project down to something very similar to what you have as your test case, I am still getting the error in the IDE.

**Comment by AlekseyTs** [comment]
@jrmoreno1 

>  I have reduced the example project down to something very similar to what you have as your test case, I am still getting the error in the IDE.

Are you referring to RemoveUnnecessaryImportsFixableConsoleCore2.zip attached above?

**Comment by jrmoreno1** [comment]
@AlekseyTs : yes, that is what I meant.  It is now all in one file, with a SDK style project, and the only references are System and System.Core.

**Comment by AlekseyTs** [comment]
I have no idea why IDE does that and I do see the behavior. Compiler doesn't report ERRID.HDN_UnusedImportStatement or ERRID.HDN_UnusedImportClause for that file. I confirmed that under debugger. If I add a different, really unused, imports, I can see that compiler reports it in that file, but only that dummy imports. Yet IDE complains on both.

At this point, I think this issue should go to IDE team for investigation.

**Comment by AlekseyTs** [comment]
@CyrusNajmabadi FYI

**Comment by jrmoreno1** [comment]
@CyrusNajmabadi : just out of curiosity, does your unit test work because you are using the release dll and AlekseyTs fail because it’s using debug? Or the difference between TestAsync and CreateCompilation?

**Comment by CyrusNajmabadi** [comment]
> Or the difference between TestAsync and CreateCompilation? 

This is the difference.  The issue exists higher than teh CreateCompilation layer.  So that's where the fix/testing had to happen as well.  Thanks :)

**Comment by AlekseyTs** [comment]
@jrmoreno1 If I understand your question correctly, we were testing different layers. I was testing compiler, which worked as expected and @CyrusNajmabadi is testing IDE, which behaved incorrectly.
